### ISSUE TITLE
[Feature Request] ConfirmDevice/RememberDevice flow for aws cognito authentication

### ISSUE THREAD

**Issue by yurii-chikh-vitech** [aws-amplify/amplify-flutter#605]
We need to add tracking device functionality to our flutter mobile app. It will be used with SMS_MFA confirmation user flow which is supported by flutter amplify_auth_cognito package. But there is no such methods as RememberDevice or ConfirmDevice.

There is support for Android and iOS for confirm device functionality
```java
Amplify.Auth.rememberDevice()
```
also there is support in aws-java-sdk-cognitoidp AWSCognitoIdentityProvider
```java
ConfirmDeviceResult confirmDevice(ConfirmDeviceRequest confirmDeviceRequest);
```

Not to use device tracking and confirmation is not a option because it's hard user experience to pass SMS_MFA code in each login. 

Any suggestions or thoughts when it will be supported by amplify-flutter?

**Comment by Jordan-Nelson** [comment]
Hi @yurii-chikh-vitech - Thanks for opening the issue. I do not have a timeline for when this will be implemented, but I am going to mark this as a feature request to track interest.

**Comment by dnys1** [comment]
This was released in version 0.2.2 with the following API:
- [`Amplify.Auth.rememberDevice`](https://pub.dev/documentation/amplify_flutter/latest/amplify_interface/AuthCategory/rememberDevice.html)
- [`Amplify.Auth.forgetDevice`](https://pub.dev/documentation/amplify_flutter/latest/amplify_interface/AuthCategory/forgetDevice.html)
- [`Amplify.Auth.fetchDevices`](https://pub.dev/documentation/amplify_flutter/latest/amplify_interface/AuthCategory/fetchDevices.html)
